crates/shared/src/price_estimation.rs

```diff
@@ -222,6 +214,52 @@ pub struct Arguments {
     pub quote_timeout: Duration,
 }
 
+#[derive(clap::Parser)]
+pub struct CoinGecko {
+    /// The API key for the CoinGecko API.
+    #[clap(long, env)]
+    pub coin_gecko_api_key: Option<String>,
+
+    /// The base URL for the CoinGecko API.
+    #[clap(
+        long,
+        env,
+        default_value = "https://api.coingecko.com/api/v3/simple/token_price"
+    )]
+    pub coin_gecko_url: Url,
+
+    #[clap(flatten)]
+    pub coin_gecko_buffered: Option<CoinGeckoBuffered>,
+}
+
+#[derive(clap::Parser)]
+#[clap(group(
+    clap::ArgGroup::new("buffered")
```

Comment:
One more nit I thought about after approving:
I assume these group names have to be unique and given that we already have a buffering system for rpc requests I would preemptively like to disambiguate them by calling this `coingecko_buffering` or sth like that.

Reply:
great idea, changed it to `coin_gecko_buffered` to match the field name too :ok_hand: